overlord/snapstate/snapstate.go

```diff
@@ -613,16 +613,26 @@ func removeInactiveRevision(st *state.State, name string, revision snap.Revision
 }
 
 // canRemove verifies that a snap can be removed.
-func canRemove(si *snap.Info, active bool) bool {
+func canRemove(si *snap.Info, snapst *SnapState, removeAll bool) bool {
+	// removing single revisions is generally allowed
+	if !removeAll {
```

Comment:
simplified the logic around removeAll (it really means either bare "snap remove" or last revision as well now)